lib/assets/javascripts/twine.coffee

```diff
@@ -46,33 +46,47 @@
     this
 
   Twine.bind = (node = rootNode, context = Twine.context(node)) ->
-    bind(context, node, true)
+    bind(context, node, getIndexesForElement(node), true)
 
   Twine.afterBound = (callback) ->
     if currentBindingCallbacks
       currentBindingCallbacks.push(callback)
     else
       callback()
 
-  bind = (context, node, forceSaveContext) ->
+  bind = (context, node, indexes, forceSaveContext) ->
     currentBindingCallbacks = []
     if node.bindingId
       Twine.unbind(node)
 
+    if defineArrayAttr = Twine.getAttribute(node, 'define-array')
+      newIndexes = defineArray(node, context, defineArrayAttr)
+      for key, value of indexes || {} when !newIndexes[key]?
+        newIndexes[key] = value
+      indexes = newIndexes
+      # register the element early because subsequent bindings on the same node might need to make use of the index
+      element = findOrCreateElementForNode(node)
+      element.indexes = indexes
+
     for type, binding of Twine.bindingTypes when definition = Twine.getAttribute(node, type)
-      element = {bindings: []} unless element  # Defer allocation to prevent GC pressure
```

Comment:
Since this is in a loop the first iteration will initialize it. It looks like it wasn't being initialized outside because of the large amount of elements in the dom that wouldn't have any bindings, so that we didn't create and then throw away the `element` object, making the GC work harder